index.js

```diff
@@ -199,10 +203,63 @@ function session(options){
 
       setcookie(res, name, req.sessionID, secrets[0], cookie.data);
     });
+    
+    var _write = res.write;
+
+    if (saveBeforeLocationHeader) {
+      var _writeHead = res.writeHead;
+      var headWritten = false;
+      res.writeHead = function writeHead(statusCode, reason, obj) {
+        if (headWritten) return;
+        headWritten = true;
 
+        // search for location header only for 3xx status codes
+        if (statusCode < 300 || statusCode >= 400) {
```

Comment:
It should be fine, implicit `writeHead` is always called with already-set response code.

https://github.com/nodejs/io.js/blob/3777f415625ce538de5edbb19f7330356da190a8/lib/_http_server.js#L150

This case is also covered by added test `should buffer the response #2`.